### PR TITLE
Refactor simplify extract

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Yo @ozten, I was trying to figure out why my integration of connect-cachify wasn't working today and ended up doing a lot of cleanup while I was trying to noodle on what various parts are doing.

Would you do a review? If an r+, mind `npm publish`ing?

Note, no tests were modified, this is a pure refactor.

Now, if I could just figure out why travis hates me.

### CODE REVIEW

File: lib/connect-cachify.js [hunk 1]
```diff
@@ -9,134 +9,194 @@ var crypto = require('crypto'),
...
+    done(do_contents_match_expected_hash(contents, expected_hash));
+  });
+};
+
+
+// clean the input options passed in by the user.
+var clean_options = function (options) {
+  options = options || {};
+  var cleaned = options;
+
```
Comment (ozten) [inline lib/connect-cachify.js:163]: Thanks for the improved names on everything.

File: lib/connect-cachify.js [hunk 2]
```diff
@@ -270,4 +374,20 @@ var escape_regex = function (str) {
...
+  } else if (typeof resp.locals === 'function') {
+    resp.locals({
+      cachify_prefetch: cachify_prefetch,
+      cachify_js: cachify_js,
+      cachify_css: cachify_css,
+      cachify: cachify
+    });
+  }
+};
+
```
Comment (ozten) [inline lib/connect-cachify.js:402]: Nice work!

**Comment by ozten** [comment]
r+

Not able to automatically merge.

We can put a release together today, and I'll upgrade to it and test it out.

**Comment by shane-tomlinson** [comment]
@ozten - thanks for checking it out! I rebased and re-pushed.

**Pull request merged**